src/app/Dashboard/DashboardLayoutConfig.tsx

```diff
@@ -0,0 +1,398 @@
+import { DownloadIcon, PencilAltIcon, PlusCircleIcon, TrashIcon, UploadIcon } from '@patternfly/react-icons';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+import { DashboardLayoutCreateModal } from './DashboardLayoutCreateModal';
+import { DashboardLayoutUploadModal } from './DashboardLayoutUploadModal';
+import { DEFAULT_DASHBOARD_NAME } from './DashboardUtils';
+
+export interface DashboardLayoutConfigProps {
+  children?: React.ReactNode;
+}
+
+export const DashboardLayoutConfig: React.FunctionComponent<DashboardLayoutConfigProps> = (_props) => {
+  const dispatch = useDispatch<StateDispatch>();
+  const context = React.useContext(ServiceContext);
+  const { t } = useTranslation();
+  const dashboardConfigs = useSelector((state: RootState) => state.dashboardConfigs);
+  const [isUploadModalOpen, setIsUploadModalOpen] = React.useState(false);
+  const [isCreateModalOpen, setIsCreateModalOpen] = React.useState(false);
+  const [isDeleteWarningModalOpen, setIsDeleteWarningModalOpen] = React.useState(false);
+  const [isSelectorOpen, setIsSelectorOpen] = React.useState(false);
+  const [oldName, setOldName] = React.useState<string | undefined>(undefined);
+  const [selectDelete, setSelectDelete] = React.useState<string>('');
+
+  const deleteRef = React.useRef<HTMLButtonElement>(null);
+
+  const currLayout = React.useMemo(() => dashboardConfigs.layouts[dashboardConfigs.current], [dashboardConfigs]);
+
+  const deletionDialogsEnabled = React.useMemo(
+    () => context.settings.deletionDialogsEnabledFor(DeleteOrDisableWarningType.DeleteDashboardLayout),
+    [context.settings]
+  );
+
+  const handleUploadModalOpen = React.useCallback(
+    (_ev: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+      setIsUploadModalOpen(true);
+    },
+    [setIsUploadModalOpen]
+  );
+
+  const handleUploadModalClose = React.useCallback(() => {
+    setIsUploadModalOpen(false);
+  }, [setIsUploadModalOpen]);
+
+  const handleDownloadLayout = React.useCallback(
+    (_ev: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+      context.api.downloadDashboardLayout(currLayout);
+    },
+    [context.api, currLayout]
+  );
+
+  const handleCreateModalOpen = React.useCallback(
+    (_ev: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+      setOldName(undefined);
+      setIsCreateModalOpen(true);
+    },
+    [setOldName, setIsCreateModalOpen]
+  );
+
+  const handleCreateModalClose = React.useCallback(() => {
+    setIsCreateModalOpen(false);
+  }, [setIsCreateModalOpen]);
+
+  const handleDeleteWarningModalOpen = React.useCallback(
+    (_ev: React.MouseEvent<HTMLButtonElement, MouseEvent>, layout: string) => {
+      setSelectDelete(layout);
+      setIsDeleteWarningModalOpen(true);
+    },
+    [setSelectDelete, setIsDeleteWarningModalOpen]
+  );
+
+  const handleDeleteWarningModalClose = React.useCallback(() => {
+    setIsDeleteWarningModalOpen(false);
```

Comment:
Reset `selectDelete`? Same for `handleDelete`?

Reply:
I think it wouldn't matter since selectDelete is always changed to before dispatch is called, so this would be extra calls that are not needed. Thoughts?

Reply:
ahh yehh no worries. Just that intermediate states can be consistent (like if there is nothing to delete, then selectDelete should be undefined)...

Reply:
Okay, if that's the guideline then I can change it.

Reply:
not exactly guideline haha but its easier to debug :D